go/multiorch/store/pooler_info.go

```diff
@@ -0,0 +1,40 @@
+// Copyright 2025 Supabase, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package store
+
+import (
+	"time"
+
+	"github.com/multigres/multigres/go/pb/clustermetadata"
+)
+
+// PoolerHealthCheckStatus represents runtime state of a MultiPooler instance.
+// This is the minimal version that stores:
+// - The MultiPooler record from topology
+// - Timestamps for staleness detection
+// - Computed fields for quick access
+type PoolerHealthCheckStatus struct {
```

Comment:
I think we will eventually need to store the grpcConn here as well. So it is more than just status, but that is for the future. We can use TabletHealth from vitess for inspiration.

Reply:
In Vitess for VTOrc, we don't store the connection this way. We use `tmcclient` that internally manages the connections. I was thinking on following that approach here as well. Let me know if you think we should revisit this

Reply:
My inclination is to use a streaming health check similar to what vtgate does with vttablets. The counter argument is that other than periodic health checks, vtorc (and now multiorch) is not constantly sending and receiving grpc messages, so why keep the connection live (with its associated keep alive overhead)? Let's discuss this in more detail offline.

Reply:
This would be part of the same discussion https://github.com/multigres/multigres/pull/240#discussion_r2525036269